syne_tune/optimizer/schedulers/searchers/regularized_evolution.py

```diff
@@ -0,0 +1,133 @@
+import copy
+
+from collections import deque
+from typing import Optional, Dict, List
+from dataclasses import dataclass
+
+from syne_tune.optimizer.schedulers.searchers import SearcherWithRandomSeed
+from syne_tune.search_space import Domain, Categorical
+
+
+@dataclass
+class PopulationElement:
+    score: int = 0
+    config: dict = None
+
+
+class RegularizedEvolution(SearcherWithRandomSeed):
+    def __init__(
+            self,
+            configspace,
+            metric: str,
+            mode: str = 'min',
+            population_size: int = 100,
+            sample_size: int = 10,
+            points_to_evaluate: Optional[List[Dict]] = None,
+            **kwargs
+    ):
+        """
+        Implements the regularized evolution algorithm proposed by Real et al. The original implementation only
+        considers categorical hyperparameters. For integer and float parameters we sample a new value uniformly
+        at random.
+
+        Real, E., Aggarwal, A., Huang, Y., and Le, Q. V.
+        Regularized Evolution for Image Classifier Architecture Search.
+        In Proceedings of the Conference on Artificial Intelligence (AAAI’19)
+
+        The code is based one the original regularized evolution open-source implementation:
+        https://colab.research.google.com/github/google-research/google-research/blob/master/evolution/regularized_evolution_algorithm/regularized_evolution.ipynb
+
+
+        Parameters
+        ----------
+        configspace: dict
+            Configuration space for trial evaluation function
+        metric : str
+            Name of metric to optimize, key in result's obtained via
+            `on_trial_result`
+        mode : str
+            Mode to use for the metric given, can be 'min' or 'max', default to 'min'.
+        population_size : int
+            Size of the population.
+        sample_size : int
+            Size of the candidate set to obtain a parent for the mutation.
+        random_seed : int
```

Comment:
Remove this comment, or otherwise add comments about random_seed_generator, random_seed in other searchers.